aip/client-libraries/4234.md

```diff
@@ -0,0 +1,110 @@
+---
+id: 4234
+state: draft
+created: 2021-02-10
+---
+
+# Common service client mixins
+
+Often, APIs can have common features for administering resources that are
+separated into utility services. These common, utility services are shared among
+the discrete product services. These common services each have a
+centrally-defined surface, but individual instances are hosted alongside each
+product service. Thus, the surfaces of each common service hosted by the product
+service are "mixed in" at runtime.
+
+
+## Guidance
+
+Client libraries **may** provide client methods for the common services declared
+to be mixed-in by a product service. Such methods improve the user experience by
+presenting the mixin methods from a client configured to communicate with the
+product service that hosts the mixin service.
+
+If client libraries support mixin services, they **must** support the following
+common services (and **may** support others):
+
+- `google.cloud.location.Locations`
+- `google.iam.v1.IAMPolicy`
+- `google.longrunning.Operations`
+
+**Note:** The list of supported mixin services and the mixin services
+themselves change infrequently. Any new common service to be supported should be
+added to the list here and generators should be updated.
+
+To be generated into a client library, a mixin service **must** be declared
+under [apis] in a [google.api.Service]. Furthermore, it **must** have
+[google.api.http] bindings declared for all of its RPCs in the [http]
+configuration of the same [google.api.Service] (more details on these bindings
+in AIP-127).
+
+
+## Implementing mixin support
+
+### Generator configuration
+
+Client library generators **must** accept the file path of a
+[google.api.Service] in YAML form. This flag **must** be optional - meaning
+the flag **must not** be required for basic client generation that omits the
+mixin methods.
+
+### Mixin API client configuration
+
+If a mixin API utilizes client library configuration such as
+`google.api.method_signature` or request header injection, generators **may**
+support it in the generated mixin methods if the configuration is accessible.
```

Comment:
I'm not sure how to be more explicit because we both proto annotations and files used for configuration. I guess in this sentence I mean annotations. WDYT?

Reply:
I've made this paragraph more explicit.